sisyphus/hash.py

```diff
@@ -152,6 +152,9 @@ def sis_hash_helper(obj):
         assert obj.__name__ != "<lambda>", "Hashing of lambda functions is not supported"
         assert obj.__module__ != "__main__", "Hashing of functions defined in __main__ is not supported"
         byte_list.append(sis_hash_helper((obj.__module__, obj.__qualname__)))
+    elif ismethod(obj):
+        # Handle bound methods
+        byte_list.append(sis_hash_helper((obj.__self__, obj.__func__)))
```

Comment:
Wouldn't it be better if already `get_object_state` can handle that? Then you don't need this here, but adapt `get_object_state`.